src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,82 @@
-import React from 'react';
+import React, { FormEvent, useState } from 'react';
+import { Comment, CommentData } from '../types/Comment';
+import classNames from 'classnames';
+import { createComment } from '../api/api';
+
+type Props = {
+  postId: number;
+  addComment: (comment: Comment) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ postId, addComment }) => {
+  const [comment, setComment] = useState<CommentData>({
+    email: '',
+    body: '',
+    name: '',
+    postId,
+  });
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isBodyError, setIsBodyError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isPostError, setIsPostError] = useState(false);
+
+  const clearAllError = () => {
+    setIsLoading(false);
+    setIsBodyError(false);
+    setIsNameError(false);
+    setIsEmailError(false);
+  };
+
+  const submitComment = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!comment.name) {
+      setIsNameError(true);
+    }
+
+    if (!comment.email) {
+      setIsEmailError(true);
+    }
+
+    if (!comment.body) {
+      setIsBodyError(true);
+    }
+
+    if (!comment.name || !comment.email || !comment.body) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    createComment(comment)
+      .then(result => {
+        addComment({ ...comment, id: result.id });
+        setComment(prev => ({ ...prev, body: '' }));
+      })
+      .catch(() => {
+        setIsPostError(true);
+        setTimeout(() => {
+          setIsPostError(false);
+        }, 3000);
```

Comment:
The error handling here is basic and only sets a generic error state. Consider providing more detailed feedback to the user about what went wrong, such as displaying specific error messages based on the error type.